### DISCUSSION TITLE
Export Make and Model of camera and lens as separate columns to CVS

### DISCUSSION THREAD

**Discussion by cezary-zemis** [tommi1hirvonen/ExifNotes#33] [Ideas]
When exporting roll data to CSV, camera make and lens make are concatenated with a respective name into "Camera" and "Lens" fields. As they may contain spaces (e.g. 'L. & A  Boulade frères, Lyon', 'Chambre de Voyage 27×33 cm'), it is not possible to split it automatically.

I suggest keeping "Camera" and "Lens" field as they are to maintain compatibility and adding (somehow redundant) fields "Camera make", "Camera model", "Lens make" and "Lens model".

**Comment by tommi1hirvonen** [comment]
I agree with keeping the current fields as they are for backwards compatibility and possibly handling this by adding new fields for the camera and lens make and model.

I'd be interested in hearing what are some of the use cases for the csv exports. I haven't utilized them myself, so I have no idea what workflows user's might have created based on the csv files. Also, have you tried out the JSON exports added in the latest release (v1.19.0)? I think the JSON format is actually better suited for this type of data and might work even better for various automation and scripting tasks.

**Reply by AngelaDMerkel** [reply]
Exiftool supports reading and writing from CSV, and could be used to read from the CSV to write images instead of pasting the list into the terminal as we do now. This wouldn't change my workflow considerably but it would make it easier to write to multiple different folders at once. It would be pretty easy to write a loop and iterate directories writing the images contained within with the relevant CSV file data. A basic, clumsy example:

```
#!/bin/bash

for folder in */; do
    if [ -f "${folder}*.csv" ]; then
        csv_file="${folder}*.csv"
        exiftool -csv="${csv_file}" -overwrite_original -ext tif -ext tiff "${folder}"
    fi
done
```

**Comment by cezary-zemis** [comment]
JSON output looks good, the only thin is I would re-write my script in a language that supports parsing JSON, e.g. in Python...
